### ISSUE TITLE
2D microscopy tutorial broken: No subject files found.

### ISSUE THREAD

**Issue by hermancollin** [ivadomed/ivadomed#1275]
## Issue description
The 2-class 2D microscopy tutorial [here](https://ivadomed.org/tutorials/two_class_microscopy_seg_2d_unet.html) doesn't work anymore on the master branch. I get the following error:

```
2023-01-23 12:02:11.020 | WARNING  | ivadomed.loader.bids_dataframe:create_bids_dataframe:135 - No subject files were found in 'data_axondeepseg_sem' dataset. Skipping dataset.
Traceback (most recent call last):
  File "/home/herman/.conda/envs/ivadomed_venv/bin/ivadomed", line 33, in <module>
    sys.exit(load_entry_point('ivadomed', 'console_scripts', 'ivadomed')())
  File "/home/herman/Documents/NEUROPOLY_21/ivadomed/ivadomed/main.py", line 623, in run_main
    run_command(context=context,
  File "/home/herman/Documents/NEUROPOLY_21/ivadomed/ivadomed/main.py", line 398, in run_command
    bids_df = BidsDataframe(loader_params, path_output, derivatives=True,
  File "/home/herman/Documents/NEUROPOLY_21/ivadomed/ivadomed/loader/bids_dataframe.py", line 72, in __init__
    self.create_bids_dataframe()
  File "/home/herman/Documents/NEUROPOLY_21/ivadomed/ivadomed/loader/bids_dataframe.py", line 147, in create_bids_dataframe
    raise RuntimeError("No subject files found. Check selection of parameters in config.json"
RuntimeError: No subject files found. Check selection of parameters in config.json and datasets compliance with BIDS specification.

```

### Steps to reproduce
Follow the 2-class microscopy tutorial. Error happens when we run the training config.

### System description
Linux

**Comment by hermancollin** [comment]
Last time I did training with ivadomed was a few months ago, so I knew this bug was recently introduced. After a quick investigation, it seems the problem started with this commit 5b2e3131: _Remove force indexing of microscopy and update ct (#1203)_.

I'll investigate further but in the meantime, I'd like to know if this problem is isolated or if someone can reproduce by simply trying the microscopy 2D tutorial on the master branch. We only want to see if the training starts (more specifically if the dataset is properly loaded), so it should take less than 15 minutes to assess.

**Comment by hermancollin** [comment]
Update: microscopy was indeed integrated in `pybids` 0.15.2 (as mentioned by Marie-Hélène in #1203) but the `bids-validator` package only supports it since 1.9.0. I had version 1.7.2 installed in my environment. Re-installing the whole environment did not help; I had to manually update bids-validator myself. This kind of pip sorcery is always annoying... I assumed deleting the environment and re-installing from scratch would install the latest versions available for all dependencies, but it doesn't. (bids-validator is a dependency of pybids and they didn't pin a version) 

TL;DR Fixed by updating bids-validator to >1.9.0

**Comment by GrimmSnark** [comment]
Hello,

So I am getting the same error as you but with BIDS_validator -1.10.0. Has anyone else had the same issue?

It defintely looks to be an update thing as I tried to run a previously working training folder and got the same error there too. 

Best,

Michael

**Comment by GrimmSnark** [comment]
Hello @hermancollin, So running `pip list | grep bids`

```
bids-validator          1.10.0
pybids                  0.15.5
```

But I still get the same error as before

```
2023-02-07 08:51:06.122 | WARNING  | ivadomed.loader.bids_dataframe:create_bids_dataframe:135 - No subject files were found in 'drive/MyDrive/microglia_Ivadomed/testBIDS' dataset. Skipping dataset.
Traceback (most recent call last):
  File "/usr/local/bin/ivadomed", line 8, in <module>
    sys.exit(run_main())
  File "/usr/local/lib/python3.7/site-packages/ivadomed/main.py", line 628, in run_main
    overlap_2d=args.overlap_2d if args.overlap_2d else None)
  File "/usr/local/lib/python3.7/site-packages/ivadomed/main.py", line 399, in run_command
    split_method=context.get(ConfigKW.SPLIT_DATASET).get(SplitDatasetKW.SPLIT_METHOD))
  File "/usr/local/lib/python3.7/site-packages/ivadomed/loader/bids_dataframe.py", line 72, in __init__
    self.create_bids_dataframe()
  File "/usr/local/lib/python3.7/site-packages/ivadomed/loader/bids_dataframe.py", line 147, in create_bids_dataframe
    raise RuntimeError("No subject files found. Check selection of parameters in config.json"
RuntimeError: No subject files found. Check selection of parameters in config.json and datasets compliance with BIDS specification.

```

Also the BIDS folder and config file for my attempt can be found here.

https://we.tl/t-5iVIL5wl9t

Thank you again for all your help and prompt replies!

Best,

Michael
